Sources/XMTP/Messages/MessageV2.swift

```diff
@@ -27,6 +27,17 @@ extension MessageV2 {
 			let decrypted = try Crypto.decrypt(keyMaterial, message.ciphertext, additionalData: message.headerBytes)
 			let signed = try SignedContent(serializedData: decrypted)
 
+			guard signed.sender.hasPreKey, signed.sender.hasIdentityKey else {
+				throw MessageV2Error.decodeError("missing sender pre-key or identity key")
+			}
+
+			let senderPreKey = try PublicKey(signed.sender.preKey)
+			let senderIdentityKey = try PublicKey(signed.sender.identityKey)
+
+			if !(try senderPreKey.signature.verify(signedBy: senderIdentityKey, digest: signed.sender.preKey.keyBytes)) {
```

Comment:
This is a bit confusing since we're passing `keyBytes` as the `digest` instead of a SHA256 hash. That's because our underlying crypto library always SHA256's whatever data is sent to it for this.

Reply:
I'm actually going to add this a comment in the code because this isn't the first time this has bit me.